common/cosmwasm-smart-contracts/vesting-contract/src/lib.rs

```diff
@@ -42,6 +44,48 @@ impl PledgeData {
     }
 }
 
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub enum PledgeCap {
+    Percent(u8),
+    Absolute(Uint128),
+}
+
+impl FromStr for PledgeCap {
+    type Err = String;
+
+    fn from_str(cap: &str) -> Result<Self, Self::Err> {
+        match cap.parse::<f32>() {
```

Comment:
wouldn't the validator complain about that float operation here?

Reply:
Crap, good call, yeah it will probably, altough that can never be executed in wasm, but yeah, moving to Decimal will help that as well